src/test/java/com/appdynamics/extensions/webspheremq/metricscollector/InquireChannelCmdCollectorTest.java

```diff
@@ -0,0 +1,130 @@
+        queueManager = mapper.convertValue(((List)configMap.get("queueManagers")).get(0), QueueManager.class);
+        Map<String, Map<String, WMQMetricOverride>> metricsMap = WMQUtil.getMetricsToReportFromConfigYml((List<Map>) configMap.get("mqMetrics"));
+        Map<String, WMQMetricOverride> channelMetrics = metricsMap.get(Constants.METRIC_TYPE_CHANNEL);
+        Map<String, Map<String, WMQMetricOverride>> metricsByCommand = new HashMap<>();
+        for (String key : channelMetrics.keySet()) {
+            WMQMetricOverride wmqOverride = channelMetrics.get(key);
+            String cmd = wmqOverride.getIbmCommand() == null ? "MQCMD_INQUIRE_CHANNEL_STATUS" : wmqOverride.getIbmCommand();
+            metricsByCommand.putIfAbsent(cmd, new HashMap<>());
+            metricsByCommand.get(cmd).put(key, wmqOverride);
+        }
+        channelMetricsToReport = metricsByCommand.get("MQCMD_INQUIRE_CHANNEL");
+        pathCaptor = ArgumentCaptor.forClass(List.class);
+    }
+
+    @Test
+    public void testProcessPCFRequestAndPublishQMetricsForInquireQStatusCmd() throws Exception {
+        when(pcfMessageAgent.send(any(PCFMessage.class))).thenReturn(createPCFResponseForInquireChannelCmd());
+        classUnderTest = new InquireChannelCmdCollector(channelMetricsToReport, monitorContextConfig, pcfMessageAgent, queueManager, metricWriteHelper);
+        classUnderTest.publishMetrics();
+        verify(metricWriteHelper, times(1)).transformAndPrintMetrics(pathCaptor.capture());
+        List<String> metricPathsList = Lists.newArrayList();
+        metricPathsList.add("Server|Component:Tier1|Custom Metrics|WebsphereMQ|QueueManager1|Channels|my.channel|MsgRetryCount");
+        metricPathsList.add("Server|Component:Tier1|Custom Metrics|WebsphereMQ|QueueManager1|Channels|my.channel|MsgsReceived");
+        metricPathsList.add("Server|Component:Tier1|Custom Metrics|WebsphereMQ|QueueManager1|Channels|my.channel|MsgsSent");
+
+        for (List<Metric> metricList : pathCaptor.getAllValues()) {
+            for (Metric metric : metricList) {
+                if (metricPathsList.contains(metric.getMetricPath())) {
+                    if (metric.getMetricPath().equals("Server|Component:Tier1|Custom Metrics|WebsphereMQ|QueueManager1|Channels|my.channel|MsgRetryCount")) {
+                        assertThat(metric.getMetricValue()).isEqualTo("22");
+                    }
+                    if (metric.getMetricPath().equals("Server|Component:Tier1|Custom Metrics|WebsphereMQ|QueueManager1|Channels|my.channel|MsgsReceived")) {
+                        assertThat(metric.getMetricValue()).isEqualTo("42");
+                    }
+                    if (metric.getMetricPath().equals("Server|Component:Tier1|Custom Metrics|WebsphereMQ|QueueManager1|Channels|my.channel|MsgsSent")) {
+                        assertThat(metric.getMetricValue()).isEqualTo("64");
+                    }
+                }
+            }
+        }
```

Comment:
It's poor practice to only do assertions in loops that iterate over the result, because it gives a false sense of security. For instance, if the `pathCaptor` captured an empty list containing NO METRICS AT ALL then this test would happily pass. Whoops. 

It's fine for now, since this horrible practice has already been established in this code, and we will need some effort to improve it.